redisvl/utils/vectorize/text/bedrock.py

```diff
@@ -0,0 +1,201 @@
+import json
+import os
+from typing import Any, Callable, Dict, List, Optional
+
+from pydantic.v1 import PrivateAttr
+from tenacity import retry, stop_after_attempt, wait_random_exponential
+from tenacity.retry import retry_if_not_exception_type
+
+from redisvl.utils.vectorize.base import BaseVectorizer
+
+
+class BedrockTextVectorizer(BaseVectorizer):
+    """The AmazonBedrockTextVectorizer class utilizes Amazon Bedrock's API to generate
+    embeddings for text data.
+
+    This vectorizer is designed to interact with Amazon Bedrock API,
+    requiring AWS credentials for authentication. The credentials can be provided
+    directly in the `api_config` dictionary or through environment variables:
+    - AWS_ACCESS_KEY_ID
+    - AWS_SECRET_ACCESS_KEY
+    - AWS_REGION (defaults to us-east-1)
+
+    The vectorizer supports synchronous operations with batch processing and
+    preprocessing capabilities.
+
+    .. code-block:: python
+
+        # Initialize with explicit credentials
+        vectorizer = AmazonBedrockTextVectorizer(
+            model_id="amazon.titan-embed-text-v2:0",
+            api_config={
+                "aws_access_key_id": "your_access_key",
+                "aws_secret_access_key": "your_secret_key",
+                "region_name": "us-east-1"
+            }
+        )
+
+        # Initialize using environment variables
+        vectorizer = AmazonBedrockTextVectorizer()
+
+        # Generate embeddings
+        embedding = vectorizer.embed("Hello, world!")
+        embeddings = vectorizer.embed_many(["Hello", "World"], batch_size=2)
+    """
+
+    _client: Any = PrivateAttr()
+
+    def __init__(
+        self,
+        model_id: str = "amazon.titan-embed-text-v2:0",
```

Comment:
Let's use the same init signature as other vectorizers, `model` instead of `model_id`.